packages/fiori-generator-shared/src/environment.ts

```diff
@@ -10,3 +13,16 @@ export function isCli(): boolean {
         return false;
     }
 }
+
+/**
+ * Determine if the current prompting environment is cli or a hosted extension (app studio or vscode).
+ *
+ * @returns the platform name and technical name
+ */
+export function getHostEnvironment(): { name: string; technical: string } {
```

Comment:
Doesnt this already exist elsewhere in open source?

Reply:
It's in odata service inquirer. But I attempted to make updates to this before and you didn't want contributions in the module yet.

Reply:
This is ok since its not impacting the code thats under development directly, its a utility. Please go ahead and update, thx @cianmSAP

Reply:
So the function that has been added is not doing the same thing and will return different results.